auth0/v3/authentication/token_verifier.py

```diff
@@ -0,0 +1,333 @@
+                'Audience (aud) claim must be a string or array of strings present in the ID token')
+
+        if type(payload['aud']) is list and not opt['aud'] in payload['aud']:
+            payload_audiences = ", ".join(payload['aud'])
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the ID token; expected "{}" but was '
+                'not one of "{}"'.format(opt['aud'], payload_audiences))
+        elif type(payload['aud']) == str and payload['aud'] != opt['aud']:
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the ID token; expected "{}" '
+                'but found "{}"'.format(opt['aud'], payload['aud']))
+
+        # --Time validation (epoch)--
+        now = opt['_clock'] or time.time()
+        leeway = opt['leeway']
+
+        # Expires at
+        if 'exp' not in payload or type(payload['exp']) != int:
+            raise TokenValidationError('Expiration Time (exp) claim must be a number present in the ID token')
+
+        exp_time = payload['exp'] + leeway
+        if now > exp_time:
+            raise TokenValidationError(
+                'Expiration Time (exp) claim error in the ID token; current time ({}) is '
+                'after expiration time ({})'.format(now, exp_time))
+
+        # Issued at
+        if 'iat' not in payload or type(payload['iat']) != int:
+            raise TokenValidationError('Issued At (iat) claim must be a number present in the ID token')
+
+        iat_time = payload['iat'] - leeway
+        if now < iat_time:
+            raise TokenValidationError(
+                'Issued At (iat) claim error in the ID token; current time ({}) is '
+                'before issued at time ({})'.format(now, iat_time))
+
+        # Nonce
+        if 'nonce' in opt and opt['nonce']:
+            if 'nonce' not in payload or type(payload['nonce']) != str:
+                raise TokenValidationError('Nonce (nonce) claim must be a string present in the ID token')
+            if payload['nonce'] != opt['nonce']:
+                raise TokenValidationError(
+                    'Nonce (nonce) claim mismatch in the ID token; expected "{}", '
+                    'found "{}"'.format(opt['nonce'], payload['nonce']))
+
+        # Authorized party
+        if type(payload['aud']) is list and len(payload['aud']) > 1:
+            if 'azp' not in payload or type(payload['azp']) != str:
+                raise TokenValidationError(
+                    'Authorized Party (azp) claim must be a string present in the ID token when '
+                    'Audience (aud) claim has multiple values')
+            if payload['azp'] != opt['aud']:
+                raise TokenValidationError(
+                    'Authorized Party (azp) claim mismatch in the ID token; expected "{}", '
+                    'found "{}"'.format(opt['aud'], payload['azp']))
+
+        # Authentication time
+        if 'max_age' in opt and opt['max_age']:
```

Comment:
`'max_age' in opt` will always be truthy

This would be simpler if you were just using the `max_age` argument